### PR TITLE
[NPU] Fix SDPA decomposition with boolean attention mask

### PR DESCRIPTION
**Pull request by AsyaPronina**
### Details:
 - *Seems like there was redudant inverse operation for incoming boolean attention_mask in SDPA decompose trasnformation. (Or might be mask was coming inversed before?)*
 - *Attention mask is passed where `True` values mean that we need to attend to these tokens. `Select` op will use these `True` values of attention mask to select elements from first argument: https://docs.openvino.ai/2025/api/c_cpp_api/classov_1_1op_1_1v1_1_1_select.html*

### Tickets:
 - *EISW-180454*

### CODE REVIEW

**Comment by dmatveev** [comment]
Is it safe to merge, have we got the nightly tests here?

**Comment by AsyaPronina** [comment]
> Is it safe to merge, have we got the nightly tests here?

Yes, we've got them recently and PR is safe to merge!
